### DISCUSSION TITLE
Logowanie za pomocą AAD w mvc

### DISCUSSION THREAD

**Discussion by szympulka** [AzureCommunityPL/group#21] [Pomoc] [answered]
Hej,

Próbuję dodać logowanie użytkownika z domeny firmowej do aplikacji. Stworzyłem próbną aplikację w AAD i wygenerowałem gotowy kod do testów, wszystko działa. Skopiowałem zatem 1:1 to co zostało tam zastosowane i wkleiłem do mojego projektu. W AAD podałem localhost, w apce podmieniłem klucze i AAD mnie zalogowało (chyba). Widać, że idzie bardzo dużo przekierowań, po czym dostaję komunikat: "We couldn't sign you in. Please try again." Czy mogę jakkolwiek dowiedzieć się, dlaczego nie udało się zalogować? 

![image](https://user-images.githubusercontent.com/12598823/90779343-0021b100-e2fe-11ea-95a7-7fab5343f4cf.png)

**Comment by PoborcaMaciej** [comment]
A w signinach od strony aplikacji nic nie widac ? (AAD -> Enterprise Application -> Sigins)

**Reply by szympulka** [reply]
tak, dostaję Status Success

**Comment by kamil-mrzyglod** [comment]
Zobacz pasek adresu czy przypadkiem nie masz tam pełnej informacji co poszło nie tak.

**Comment by tonyszko** [comment]
W AAD masz log który Ci powie co się dzieje - jeżeli masz response to tam też powinien być kod błędu, który powie Ci co się dzieje. 

BTW - a uaktualniłeś definicję aplikacji w AAD (signin URL itp?) Generalnie copy&paste często powoduje różnego rodzaju 'nie skopiowane' zmiany.

**Reply by szympulka** [reply]
@tonyszko właśnie co do URL. Podałem adres wygenerowany + dodałem swojego localhosta. Efekt był taki, że i w generowanym kodzie jak i moim logowanie nie działało. Nie mogę dodać więcej adresów?
![image](https://user-images.githubusercontent.com/12598823/91036900-6c572a00-e608-11ea-9b30-1022ed471353.png)

**Comment by szympulka** [comment]
@kamil-mrzyglod w pasku oraz w odpowiedzi poza ciągiem znaków nic nie znalazłem. MS odpowiada mi cały czas kodem 200 
![image](https://user-images.githubusercontent.com/12598823/91038337-e4beea80-e60a-11ea-94eb-a86edc83bf46.png)

Wątpię ale czy to, że testowy kod jest napisany w net core 2.1 a ja piszę 3.1 może mieć znaczenie?

**Reply by kamil-mrzyglod** [reply]
Z mojego doświadczenia może - miałem sytuacje kiedy paczki w wersji X działały a ich zmiana na wersję Y wywalała proces uwierzytelniania. Generalnie masz tutaj różnice major wersji - teoretycznie pomiędzy nimi kompatybilności nie ma. Skąd masz ten testowy kod?

**Reply by szympulka** [reply]
Generowany z portalu Azure bezpośrednio

**Reply by wi5nia** [reply]
Jestesmy na GitHubie - wrzuc gdzies ten kod i daj linka - spojrzy sie na niego skoro to taki demo kod 😃

**Reply by kamil-mrzyglod** [reply]
Heh, nawet nie zwróciłem uwagi na to, że można gdzieś sobie taki kod wygenerować. Tak jak Tomek napisał - zrób Gista albo spakuj i wrzuć do wątku to może ktoś spróbuje u siebie to odpalić.

**Reply by szympulka** [reply]
Dobra, jutro przygotuję paczkę

**Reply by mgrabarz** [reply]
Co ten screen pokazuje @szympulka , jakieś http 200 na SVG. Nie ma to zbyt wiele wspólnego z AAD. 

Przechodzisz Authorization Code Granta, tak? Na którym etapie się wywala, na GET, czy POST?

**Reply by szympulka** [reply]
Chodziło o 2 pierwsze requesty. Fakt, svg tutaj nie ma nic wspólnego. Zwyczajny przypadek

**Comment by szympulka** [comment]
https://github.com/szympulka/AAD

Wrzuciłem do repo 2 puste projekty. Jeden wygenerowany przez Poratl z net core 2.2 oraz drugi z net core 3.1. W przypadku 3.1 nie działa, choć ustawianie wydają się takie same

**Comment by szympulka** [comment, marked as answer]
Problem rozwiązałem. Wygenerował sobie nowy projekt przez VisualStudio z autoryzacją przez AAD. Tutaj copy-paste zadziałało poprawnie. Kod również był okrojony w tym przypadku

**Reply by kamil-mrzyglod** [reply]
Czyli jakie mamy rozwiązanie? Nie działało bo generowałeś dla 2.2 a jak poleciałeś z szablonem z VS na 3.1 to pykło?

**Reply by szympulka** [reply]
Dokładnie, różnica była w Startup.cs. W net.core 3.1 jest okrojony serwis. Prawdopodobnie to on gdzieś na poziomie middleware robił przekierowania w pętli. Stąd też w AAD miałem informację, że zalogowane poprawnie
